flink-core/src/main/java/org/apache/flink/util/WrappingProxyUtil.java

```diff
@@ -31,15 +33,19 @@ private WrappingProxyUtil() {
 	}
 
 	@SuppressWarnings("unchecked")
-	public static <T> T stripProxy(T object) {
+	public static <T> T stripProxy(@Nullable final WrappingProxy<T> wrappingProxy) {
+		if (wrappingProxy == null) {
+			return null;
+		}
 
-		T previous = null;
+		WrappingProxy<T> previous = null;
+		Object delegate = wrappingProxy.getWrappedDelegate();
 
-		while (object instanceof WrappingProxy && previous != object) {
-			previous = object;
-			object = ((WrappingProxy<T>) object).getWrappedDelegate();
+		while (delegate instanceof WrappingProxy && previous != delegate) {
```

Comment:
Why is the check `previous != delegate` needed?

Reply:
Otherwise you can generate an endless-loop. See FLINK-7524.

Reply:
As discussed offline, max iterations is 128 now.